### PR TITLE
Add autofac DI tests

### PR DESCRIPTION
**Pull request by xiazen**
both internal and external customer has reported they are using Autofac with Orleans. In theory Orleans should work with Autofac, after fix #4453 , I just want to ensure it really works,by adding DI tests running against Autofac DI container.

### CODE REVIEW

File: test/DependencyInjection.Tests/Autofac/DependencyInjectionDisambiguationTestsUsingAutofac.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+using Autofac.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace DependencyInjection.Tests.Autofac
+{
+
+    [TestCategory("DI"), TestCategory("Functional")]
+    public class DependencyInjectionDisambiguationTestsUsingAutofac : DependencyInjectionDisambiguationTestRunner
+    {
+        protected override IServiceProvider BuildeServiceProvider(IServiceCollection services)
```
Comment (ReubenBond) [inline test/DependencyInjection.Tests/Autofac/DependencyInjectionDisambiguationTestsUsingAutofac.cs]: typo on build

**Comment by sergeybykov** [comment]
@dotnet-bot test bvt
@dotnet-bot test functional

**Comment by xiazen** [comment]
yoyo, merge it ? I fixed the build error

**Comment by ReubenBond** [comment]
yoyo thanks for the reminder :)

**Pull request merged**